tests/Backend/ReaderWorkspaces/WorkspacesReaderSharingTest.php

```diff
@@ -0,0 +1,238 @@
+<?php
+
+namespace Backend\ReaderWorkspaces;
+
+use Doctrine\DBAL\Connection;
+use Keboola\Csv\CsvFile;
+use Keboola\StorageApi\Client;
+use Keboola\StorageApi\Components;
+use Keboola\StorageApi\Options\Components\Configuration;
+use Keboola\StorageApi\Options\Components\ListComponentsOptions;
+use Keboola\StorageApi\WorkspaceLoginType;
+use Keboola\StorageApi\Workspaces;
+use Keboola\TableBackendUtils\Escaping\Snowflake\SnowflakeQuote;
+use Keboola\Test\Backend\WorkspaceConnectionTrait;
+use Keboola\Test\Backend\WorkspaceCredentialsAssertTrait;
+use Keboola\Test\Backend\Workspaces\Backend\WorkspaceBackendFactory;
+use Keboola\Test\StorageApiTestCase;
+use Keboola\Test\Utils\PemKeyCertificateGenerator;
+use Keboola\Test\Utils\SnowflakeConnectionUtils;
+
+class WorkspacesReaderSharingTest extends StorageApiTestCase
```

Comment:
musel jsem to oddelit, protoze WorkspacesReaderTest potrebuje dedit z WorkspacesTestCase , kde se ale vyrabi buckety a to ja nepotrebuju